### PR TITLE
workstyle: unstable-2021-05-09 -> unstable-2023-08-23

### PR DESCRIPTION
**Pull request by pinpox**
…d3753

Diff: https://github.com/pierrechevalier83/workstyle/compare/f2023750d802259ab3ee7d7d1762631ec157a0b1...8bde72d9a9dd67e0fc7c0545faca53df23ed3753

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pinpox** [comment]
https://github.com/NixOS/nixpkgs/issues/240005

**Review by natsukium — CHANGES_REQUESTED** [review]
Please update rev and hash.

File: pkgs/applications/window-managers/i3/workstyle.nix [hunk 1]
```diff
@@ -5,16 +5,26 @@
...
-  cargoSha256 = "0xwv8spr96z4aimjlr15bhwl6i3zqp7jr45d9zr3sbi9d8dbdja2";
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "swayipc-3.0.1" = "sha256-3Jhz3+LhncSRvo3n7Dh5d+RWQSvEff9teuaDZLLLEHk=";
+    #   "flash-lso-0.5.0" = "sha256-9uH3quxRzLtmHJs5WF/GRxWkXL/KFyOl182HKcHNnuc=";
+    #   "gc-arena-0.2.2" = "sha256-/H9VcTesBD+IA7bUf208b0HQ/cIUDAz9TJBBywf6akA=";
+    #   "h263-rs-0.1.0" = "sha256-4kBg09VHyiQTvUbvcTb5g/BVcOpRFZ1fVEuRWXv5XwE=";
+    #   "nellymoser-rs-0.1.2" = "sha256-GykDQc1XwySOqfxW/OcSxkKCFJyVmwSLy/CEBcwcZJs=";
+    #   "nihav_codec_support-0.1.0" = "sha256-rE9AIiQr+PnHC9xfDQULndSfFHSX4sqKkCAQYVNaJcQ=";
```
Comment (natsukium) [inline pkgs/applications/window-managers/i3/workstyle.nix:25]:
```suggestion
```
Please remove unused comments.

File: pkgs/applications/window-managers/i3/workstyle.nix [hunk 2]
```diff
@@ -5,16 +5,26 @@
...
     repo = pname;
-    rev = "f2023750d802259ab3ee7d7d1762631ec157a0b1";
-    sha256 = "04xds691sw4pi2nq8xvdhn0312wwia60gkd8b1bjqy11zrqbivbx";
+    rev = "8bde72d9a9dd67e0fc7c0545faca53df23ed3753";
+    sha256 = "sha256-yhnt7edhgVy/cZ6FpF6AZWPoeMeEKTXP+87no2KeIYU=";
   };
 
-  cargoSha256 = "0xwv8spr96z4aimjlr15bhwl6i3zqp7jr45d9zr3sbi9d8dbdja2";
+  cargoLock = {
+    lockFile = ./Cargo.lock;
```
Comment (natsukium) [inline pkgs/applications/window-managers/i3/workstyle.nix:18]:
```suggestion
    lockFile = ./workstyle-Cargo.lock;
```
I'm not familiar with the i3 ecosystem, but there are many subpackages in the i3 directory.
Could you change this to something like workstyle-Cargo.lock, as other packages may also need Cargo.lock?

Comment (pinpox) [reply pkgs/applications/window-managers/i3/workstyle.nix]: Good idea - file is moved.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 250962` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>workstyle</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
I will squash merge this time, but it would be nice if you could rebase the commits into one as much as possible.

**Pull request merged**